### PR TITLE
added jscs linter (closes #66)

### LINKED ISSUE #66: Adopt a style guide
I think it's a good idea for Isomer to adapt a specific code style guide. Since Gulp is already used as a build tool, we can maybe add a JS linter stream, along with a config file that defines what's allowed and what's not. I notice the code that people write in their forks of Isomer is not styled in the same way as master.

### PR DESCRIPTION
**Pull request by whoeverest**

### CODE REVIEW

File: .jscsrc [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        "else",
+        "while",
+        "catch"
+    ],
+    "requireSpaceAfterKeywords": true,
+    "requireSpaceBeforeObjectValues": true,
+    "requireSpaceBetweenArguments": true,
+    "requireSpacesInConditionalExpression": true,
+    "requireSpacesInFunction": {
+        "beforeOpeningRoundBrace": true,
```
Comment (jdan) [inline .jscsrc]: Set this to `false` please! In fact, I'd like to make sure there *isn't* a space before the opening round brace - will "false" do that?

That's just my style :)

File: README.md [hunk 2]
```diff
@@ -106,6 +106,26 @@ listening on http://localhost:9966/
...
+The code in Isomer conforms to some styling conventions. To run the linter, run `gulp lint`:
+
+```
+> gulp lint
+[19:09:04] Using gulpfile ~/playground/isomer/gulpfile.js
+[19:09:04] Starting 'lint'...
+[19:09:04] 'lint' errored after 160 ms
+[19:09:04] Error in plugin 'gulp-jscs'
+Message:
+    Missing space before opening round brace at isomer.js :
```
Comment (jdan) [inline README.md]: And after that change we'd need to either replace this example or just remove it

**Comment by jdan** [comment]
Wow, awesome work - thanks! Just left a comment about `function(blah) {}` vs `function (blah) {}`

File: .jscsrc (hunk 1 above)
Comment (whoeverest) [inline .jscsrc]: Okay, sure, I prefer that style too -> `function(arg1, arg2) {}` too. But all of the functions were written with an empty space before the round bracket, so I just played along. :-)

I'll change it now.

Comment (jdan) [inline .jscsrc]: Oh lol this is totally on me. I guess that was my style back when I first wrote this! - ick!

Thanks for going along with it, but I think I'd prefer if we skipped the spacing. What do you think?

Comment (whoeverest) [inline .jscsrc]: I'd skip the spacing too.

On Sun, Mar 29, 2015 at 8:33 PM, Jordan Scales <notifications@github.com>
wrote:

> In .jscsrc <https://github.com/jdan/isomer/pull/68#discussion_r27357102>:
>
> > +    "requireCommaBeforeLineBreak": true,
> > +    "requireSemicolons": true,
> > +
> > +    // spaces
> > +    "requireSpaceBeforeBlockStatements": true,
> > +    "requireSpaceBeforeKeywords": [
> > +        "else",
> > +        "while",
> > +        "catch"
> > +    ],
> > +    "requireSpaceAfterKeywords": true,
> > +    "requireSpaceBeforeObjectValues": true,
> > +    "requireSpaceBetweenArguments": true,
> > +    "requireSpacesInConditionalExpression": true,
> > +    "requireSpacesInFunction": {
> > +        "beforeOpeningRoundBrace": true,
>
> Oh lol this is totally on me. I guess that was my style back when I first
> wrote this! - ick!
>
> Thanks for going along with it, but I think I'd prefer if we skipped the
> spacing. What do you think?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jdan/isomer/pull/68/files#r27357102>.
>



-- 
Андреј Трајчевски / 078/426-717 / andrejtrajchevski@gmail.com

**Comment by whoeverest** [comment]
@jdan Can you review once again?

**Comment by jdan** [comment]
Yep will do in a few hours when I'm back! Thanks for the ping



--



jordan scales

http://jordanscales.com

**Comment by jdan** [comment]
Really great work! Let's talk about getting something like Travis setup with this?

I'm also gonna add you as a collaborator (currently it's only me), which is something I've been meaning to do with all PR authors :) No pressure of course, but if you're interested in closing issues, fixing stuff, etc, this will make it easier.

**Comment by whoeverest** [comment]
Thanks! :-))

Sure, I'll open another issue for Travis

**Pull request merged**